### ISSUE TITLE
Autocomplete cell type does not update the list of choices correctly on Firefox

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#7570]
### Description

**Important**: this issue is not replicable on Chrome nor Edge.

### Steps to reproduce

1. Go to the first cell in the last column and open the editor
2. Type `bla`

![bla_](https://user-images.githubusercontent.com/10757813/108493271-6932b300-72a6-11eb-8d1b-0c9eef0eec30.gif)

**Result**
![image](https://user-images.githubusercontent.com/10757813/108491244-f45e7980-72a3-11eb-82d2-e1386de2415b.png)

### Demo

https://handsontable.com/docs/8.3.1/demo-autocomplete.html - current (broken)
https://handsontable.com/docs/7.4.0/demo-autocomplete.html - first broken
https://handsontable.com/docs/7.3.0/demo-autocomplete.html - last working

### Your environment
* Handsontable version: 7.4.0+
* Browser Name and version: Firefox 85.0.2
* Operating System: Windows 10

Update [forum](https://forum.handsontable.com/t/autocomplete-bug-on-firefox-all-version/5016)

**Comment by aninde** [comment]
This issue cannot be reproduced on macOS, so I'm adding label Windows.
Tested with latests macOS Big Sur 11.2.1 and older Catalina 10.15.7.

On Safari this bug can not be reproduced due to dhttps://github.com/handsontable/handsontable/issues/7413.

**Comment by AMBudnik** [comment]
We've just tested the scenario on multiple devices with @Osky772 and @wszymanski but apparently, this issue is connected to other more specific settings than a browser version of the system version. It makes it hard to fix and test.

More importantly, this issue has been reproduced also using Handsontable v2.0.0 so I'm renaming the table and removing the 'regression' label.

**Comment by wszymanski** [comment]
I've tested it using Browserstack on `Windows 10` + `FF 84`, `FF 85` and virtual machine on `Windows 7` + `FF 86`.

**Comment by Osky772** [comment]
I've tested it on `Windows 10` + `FF 86`

**Comment by Maxoudela** [comment]
Reproduced on Handsontable 10.0.0 and Firefox 94.0.2 (64 bits) on Windows 10

**Comment by Maxoudela** [comment]
The choices are filtered when the `onBeforeKeyDown` is triggered and `this.TEXTAREA.value` is used. It seems rather fair that Firefox is returning the value before the actual key has been pressed but other browsers are not working this way.

I've managed to overcome temporarily the issue by adding a small debounce on Firefox. When `queryChoices` is called, I just do nothing and call my debounce function that will then use the `this.TEXTAREA.value` at a later moment when it will be accurate.

I believe this issue could be easily fixed by increasing the [timeOffset](https://github.com/handsontable/handsontable/blob/80dac620e9414d37e0f0b592cf3cd75dcb5e0cad/handsontable/src/editors/autocompleteEditor/autocompleteEditor.js#L467) when Firefox is used.


```
import { debounce, DebouncedFunc } from 'lodash';
import Handsontable from 'handsontable';

export default class AutocompleteEditorCustom extends Handsontable.editors.AutocompleteEditor {
  private queryChoicesDebounced: DebouncedFunc<() => void> | undefined;

  /**
   * Prepares choices list based on applied argument.
   *
   * @private
   * @param {string} query The query.
   */
  queryChoices(query: string) {
    this.queryChoicesDebounced!();
  }

  open() {
    super.open();
    this.queryChoicesDebounced = debounce(() => super.queryChoices(this.TEXTAREA.value), 10);
  }

  close() {
    super.close();
    this.queryChoicesDebounced!.cancel();
    this.queryChoicesDebounced = undefined;
  }
}
```

**Comment by aninde** [comment]
This issue will be fixed for the upcoming v11.1.0
<img width="150" alt="Screenshot 2022-01-12 at 17 37 29" src="https://user-images.githubusercontent.com/30186818/149319987-a3c94a11-c8c2-407d-b9d8-559d981863b0.png">

**Comment by AMBudnik** [comment]
Issue closed as fixed in Handsontable 11.1.0
